Content.Server/Research/Components/RoboticsConsoleComponent.cs

```diff
@@ -0,0 +1,53 @@
+using Content.Server.Research.Systems;
+using Content.Shared.Radio;
+using Content.Shared.Research;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom;
+
+namespace Content.Server.Research.Components;
+
+[RegisterComponent, Access(typeof(RoboticsConsoleSystem))]
+public sealed partial class RoboticsConsoleComponent : Component
+{
+    /// <summary>
+    /// Address and data of each cyborg.
+    /// </summary>
+    [DataField]
+    public Dictionary<string, CyborgControlData> Cyborgs = new();
+
+    /// <summary>
+    /// Last time a packet was received from each cyborg.
+    /// </summary>
+    [DataField]
+    public Dictionary<string, TimeSpan> Timeouts = new();
+
+    /// <summary>
```

Comment:
This isn't going to handle persistence properly.

Reply:
ill put timeout on the data struct and have it with serializer ig